### PR TITLE
Limit the data returned by the preview handler.

### PR DESCRIPTION
**Pull request by sagarkapare**

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
LGTM. Since this doesn't have test, please merge if the code compiles locally (in the interest of time)

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java [hunk 1]
```diff
@@ -47,10 +51,12 @@
   private static final Gson GSON = new GsonBuilder().registerTypeAdapter(Schema.class,
                                                                          new SchemaTypeAdapter()).create();
   private final PreviewManager previewManager;
+  private int limit;
 
   @Inject
-  PreviewHttpHandler(PreviewManager previewManager) {
+  PreviewHttpHandler(CConfiguration cConf, PreviewManager previewManager) {
     this.previewManager = previewManager;
+    limit = cConf.getInt("preview.record.limit", 50);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java]: Just checking if this is the right place for cConf property names? Also shouldn't the default value be part of cdap-default.xml?

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java [hunk 2]
```diff
@@ -47,10 +51,12 @@
   private static final Gson GSON = new GsonBuilder().registerTypeAdapter(Schema.class,
                                                                          new SchemaTypeAdapter()).create();
   private final PreviewManager previewManager;
+  private int limit;
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java]: final?

**Comment by gokulavasan** [comment]
Couple of questions

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java (hunk 1 above)
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java]: This is not a permanent solution and it is an alpha feature. so okay not having it in cdap-default

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java (hunk 2 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java]: fixed.

**Comment by sagarkapare** [comment]
Merging.

**Pull request merged**